cargo-shuttle/src/lib.rs

```diff
@@ -267,7 +268,22 @@ impl Shuttle {
     }
 
     async fn stop(&self, client: &Client) -> Result<()> {
-        let service = client.stop_service(self.ctx.project_name()).await?;
+        let mut service = client.stop_service(self.ctx.project_name()).await?;
+
+        let progress_bar = create_spinner();
+        loop {
+            let Some(ref deployment) = service.deployment else {
+                break;
+            };
+
+            if let shuttle_common::deployment::State::Stopped = deployment.state {
+                break;
+            }
+
+            progress_bar.set_message(format!("Stopping {}", deployment.id));
+            service = client.get_service_summary(self.ctx.project_name()).await?;
+        }
+        progress_bar.finish_and_clear();
```

Comment:
Any reason for not using the current `wait_with_spinner()` helper?

Reply:
Yes, I tried to do that for a little while, but it ended up getting quite complicated. While it is generic, it's generic for project commands. So making it generic for deployer as well meant adding another generic function arg to fetch the latest state, making it generic over which type of `states_to_check` as well as response type. So we could do that, but this felt a lot simpler. I'm not sure what the best choice is, to be honest, so I'm open to changing it.

Reply:
Let's go with what you currently have. We can always add another generic version for deployer when we need it a second time